patches/corefx/0003-Enable-CoreFX-tests.patch

```diff
@@ -0,0 +1,388 @@
+diff --git a/build.proj b/build.proj
+index 09b69e0..e15d7ce 100644
+--- a/build.proj
++++ b/build.proj
+@@ -22,7 +22,8 @@
+       <_RestoreProjects Include="external\dir.proj" Build="true" />
+     </ItemGroup>
+     
+-    <ItemGroup Condition="'$(DotNetBuildFromSource)' != 'true'">
++    <!--When Building from source and including test utils then ref and src projects are to be restored-->
++    <ItemGroup Condition="('$(DotNetBuildFromSource)' != 'true') or '$(IncludeTestUtils)' == 'true'">
+       <_RestoreProjects Include="$(SourceDir)tests.proj" Build="false" />
+     </ItemGroup>
+ 
+diff --git a/external/binplacePackages/binplacePackages.depproj b/external/binplacePackages/binplacePackages.depproj
+index 48e61b9..eeb3109 100644
+--- a/external/binplacePackages/binplacePackages.depproj
++++ b/external/binplacePackages/binplacePackages.depproj
+@@ -14,8 +14,8 @@
+     </BinPlaceConfiguration>
+   </ItemGroup>
+ 
+-  <!-- Runtime / test-runtime binplacing -->
+-  <ItemGroup Condition="'$(TargetsNetStandard)' != 'true'">
++  <!-- Runtime / test-runtime binplacing, when we aren't building better live version of these libs -->
++  <ItemGroup Condition="'$(TargetsNetFx)' == 'true'">
+     <BinPlaceConfiguration Include="$(Configuration)">
+       <ItemName>BinPlaceLib</ItemName>
+       <RuntimePath>$(RuntimePath)</RuntimePath>
+diff --git a/external/netstandard/netstandard.depproj b/external/netstandard/netstandard.depproj
+index 18e54cc..5c55d91 100644
+--- a/external/netstandard/netstandard.depproj
++++ b/external/netstandard/netstandard.depproj
+@@ -81,7 +81,15 @@
+     </PropertyGroup>
+ 
+     <ItemGroup>
+-      <_NetStandard21Files Include="$(_NETStandard21RefFolder)\*.dll" />
++      <ExcludeNetStandard21Refs Include="System.Buffers" />
++      <ExcludeNetStandard21Refs Include="System.ComponentModel.Composition" />
++      <ExcludeNetStandard21Refs Include="System.Reflection.DispatchProxy" />
++      <ExcludeNetStandard21Refs Include="System.Reflection.Emit" />
++      <ExcludeNetStandard21Refs Include="System.Reflection.Emit.ILGeneration" />
++      <ExcludeNetStandard21Refs Include="System.Reflection.Emit.Lightweight" />
++      <_NetStandard21Files 
++        Include="$(_NETStandard21RefFolder)\*.dll"
++        Exclude="@(ExcludeNetStandardRefs -> '$(_NETStandard21RefFolder)\%(Identity).dll')"  />
+     </ItemGroup>
+ 
+     <Error Condition="'@(_NetStandard21Files)' == ''" Text="Could not find package assets for netstandard2.1" />
+diff --git a/src/System.ComponentModel.Composition.Registration/tests/System.ComponentModel.Composition.Registration.Tests.csproj b/src/System.ComponentModel.Composition.Registration/tests/System.ComponentModel.Composition.Registration.Tests.csproj
+deleted file mode 100644
+index 9abbcb3..0000000
+--- a/src/System.ComponentModel.Composition.Registration/tests/System.ComponentModel.Composition.Registration.Tests.csproj
```

Comment:
Could we file an issue to fix these tests?

Reply:
I'd be curious to understand what's wrong with them... First I've heard of this.

Reply:
What I observed was that [#40426](https://github.com/dotnet/corefx/pull/40426) resolved [#40417](https://github.com/dotnet/corefx/issues/40417) and the built could reach the test stage. But the System.ComponentModel.Composition still ends up being empty, so the tests which depend on this dll would fail. That's the reason I had to leave this patch intact.

Reply:
This doesn't sound right.  After that fix the built S.CM.C should be winning.  We need to take another look if it is not.

Reply:
ref package of that binary is still ~5Kb. The current state of this PR can repro it. 

`[root@20b7e9a283da source-build]# ls -lrth bin/src/corefx.1bb52e6a3db7f3673a3825f3677b9f27b9af99aa/artifacts/bin/ref/System.ComponentModel.Composition/netstandard-Release/System.ComponentModel.Composition.dll
-rwxr--r--. 5 root root 4.5K Aug 22 15:23 bin/src/corefx.1bb52e6a3db7f3673a3825f3677b9f27b9af99aa/artifacts/bin/ref/System.ComponentModel.Composition/netstandard-Release/System.ComponentModel.Composition.dll
[root@20b7e9a283da source-build]#
`